web/app/js/components/ResourceDetail.jsx

```diff
@@ -161,12 +167,24 @@ export class ResourceDetailBase extends React.Component {
     this.api.setCurrentRequests(apiRequests);
 
     Promise.all(this.api.getCurrentPromises())
-      .then(([resourceRsp, podListRsp, podMetricsRsp, upstreamRsp, downstreamRsp, edgesRsp]) => {
+      .then(results => {
+        const [resourceRsp, podListRsp, podMetricsRsp, upstreamRsp, downstreamRsp, ...rsp] = [...results];
         let resourceMetrics = processSingleResourceRollup(resourceRsp, resource.type);
         let podMetrics = processSingleResourceRollup(podMetricsRsp, resource.type);
         let upstreamMetrics = processMultiResourceRollup(upstreamRsp, resource.type);
         let downstreamMetrics = processMultiResourceRollup(downstreamRsp, resource.type);
-        let edges = processEdges(edgesRsp, this.state.resource.name);
+
+        let resourceDefinition = undefined;
```

Comment:
I'm a little uncomfortable setting the variable to `undefined` here, since if `queryForDefinition` is `false`, it will never be defined. Can we set `resourceDefinition` to `null` up in the `getInitialState()` function and avoid this line?

Reply:
Done